strawberryfields/gbs/similarity.py

```diff
@@ -201,3 +212,155 @@ def event_to_sample(photon_number: int, max_count_per_mode: int, modes: int) ->
             available_modes.remove(j)
 
     return sample
+
+
+def fac_prod(orbit: list) -> int:
+    """Computes the product of the factorial of elements in an orbit.
+
+    **Example usage**
+    >>> fac_prod([3, 2, 2, 1])
+    24
+
+    Args:
+        orbit (list[int]): the input orbit
+
+    Returns:
+        int: the product of the factorial of elements in the orbit
+    """
+
+    prod = 1
+    for i in orbit:
+        prod = prod * factorial(i)
+
+    return prod
+
+
+def compress_sample(sample: list) -> list:

```

Comment:
A very similar function `sample_to_subgraph` was actually used in `gbs.g_sample` to convert from samples to subgraphs. I decided to combine that function with this one and place in the `gbs.sample` module.